env: Unable to boot on Windows OS

### Description

It appears that version `7.0.0` of the `env` package is no longer able to start on Windows OS.

Here are all the logs after running the command:

<details><summary>Log</summary>

```
npm run wp-env start

> test@ wp-env D:\Desktop\test
> wp-env "start"

‼ Warning: could not find a .wp-env.json configuration file and could not determine if 'D:\Desktop\test' is a WordPress installation, a plugin, or a theme.
× Error while running docker-compose command.
#1 [internal] load build definition from WordPress.Dockerfile
#1 transferring dockerfile: 1.91kB 0.0s done
#1 DONE 0.1s

#2 [internal] load .dockerignore
#2 transferring context: 2B 0.0s done
#2 DONE 0.1s

#3 [internal] load metadata for docker.io/library/wordpress:latest
#3 DONE 2.5s

#4 [ 1/17] FROM docker.io/library/wordpress@sha256:f3b8d54fe9b80e88255121a46933ca1961fb9f3f39e6981a668cdb7f019a5bbd
#4 DONE 0.0s

#5 [ 9/17] RUN apt-get -qy install sudo
#5 CACHED

#6 [11/17] RUN echo 'upload_max_filesize = 1G' >> /usr/local/etc/php/php.ini
#6 CACHED

#7 [ 6/17] RUN useradd -m -u 0 -g 0 {username} || true
#7 CACHED

#8 [ 7/17] RUN apt-get -qy update
#8 CACHED

#9 [ 2/17] RUN sed -i 's|deb.debian.org/debian stretch|archive.debian.org/debian stretch|g' /etc/apt/sources.list
#9 CACHED

#10 [ 8/17] RUN apt-get -qy install autoconf            dpkg-dev                file            g++             gcc             libc-dev                make            pkg-config              re2c && touch /usr/local/etc/php/php.ini
#10 CACHED

#11 [ 5/17] RUN groupadd -g 0 {username} || true
#11 CACHED

#12 [ 3/17] RUN sed -i 's|security.debian.org/debian-security stretch|archive.debian.org/debian-security stretch|g' /etc/apt/sources.list
#12 CACHED

#13 [15/17] RUN php /tmp/composer-setup.php --install-dir=/usr/local/bin --filename=composer
#13 CACHED

#14 [12/17] RUN echo 'post_max_size = 1G' >> /usr/local/etc/php/php.ini
#14 CACHED

#15 [13/17] RUN curl -sS https://getcomposer.org/installer -o /tmp/composer-setup.php
#15 CACHED

#16 [14/17] RUN export COMPOSER_HASH=`curl -sS https://composer.github.io/installer.sig` && php -r "if (hash_file('SHA384', '/tmp/composer-setup.php') === '$COMPOSER_HASH') { echo 'Installer verified'; } else { echo 'Installer corrupt'; unlink('/tmp/composer-setup.php'); } echo PHP_EOL;"
#16 CACHED

#17 [10/17] RUN echo "{username} ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
#17 CACHED

#18 [ 4/17] RUN sed -i '/stretch-updates/d' /etc/apt/sources.list
#18 CACHED

#19 [16/17] RUN rm /tmp/composer-setup.php
#19 CACHED

#20 [17/17] RUN composer global require --dev yoast/phpunit-polyfills:"^1.0"
#20 ERROR: process "/bin/sh -c composer global require --dev yoast/phpunit-polyfills:\"^1.0\"" did not complete successfully: unable to find user {username}: no matching entries in passwd file  
------
 > [17/17] RUN composer global require --dev yoast/phpunit-polyfills:"^1.0":
------
failed to solve: process "/bin/sh -c composer global require --dev yoast/phpunit-polyfills:\"^1.0\"" did not complete successfully: unable to find user {username}: no matching entries in passwd file
npm ERR! code ELIFECYCLE
npm ERR! errno 17
npm ERR! test@ wp-env: `wp-env "start"`
npm ERR! Exit status 17
npm ERR!
npm ERR! Failed at the test@ wp-env script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\{username}\AppData\Roaming\npm-cache\_logs\2023-05-20T13_27_15_180Z-debug.log
```
</details>

I deleted all Docker containers, images, volumes, and networks before running the command. With env version `6.0.0`, it can be started normally.

The following are possible error locations in the logs:

```
#20 [17/17] RUN composer global require --dev yoast/phpunit-polyfills:"^1.0"
#20 ERROR: process "/bin/sh -c composer global require --dev yoast/phpunit-polyfills:\"^1.0\"" did not complete successfully: unable to find user {username}: no matching entries in passwd file  
------
 > [17/17] RUN composer global require --dev yoast/phpunit-polyfills:"^1.0":
------
failed to solve: process "/bin/sh -c composer global require --dev yoast/PHPUnit-polyfills:\"^1.0\"" did not complete successfully: unable to find user {username}: no matching entries in passwd file
```

`{username}` shows the actual Windows OS login user. If the following commands are executed directly, the installation completed successfully:

```
composer global require --dev yoast/phpunit-polyfills:"^1.0"
```

I think this issue might be related to #50408, where the command causing the error was added.



### Step-by-step reproduction instructions

- Create a package.json like the following:
```json
{
  "name": "test",
  "scripts": {
    "wp-env": "wp-env"
  },
  "dependencies": {
    "@wordpress/env": "^7.0.0"
  }
}
```

- `npm i`
- `npm run wp-env start`


### Screenshots, screen recording, code snippet

_No response_

### Environment info

_No response_

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

I expected the problem to be a failure to add users in the Docker container. So I made the following changes:

```diff
diff --git a/packages/env/lib/init-config.js b/packages/env/lib/init-config.js
index 4a573c97aa..a86b40896e 100644
--- a/packages/env/lib/init-config.js
+++ b/packages/env/lib/init-config.js
@@ -133,8 +133,8 @@ ARG HOST_USERNAME
 ARG HOST_UID
 ARG HOST_GID
 # When the IDs are already in use we can still safely move on.
-RUN groupadd -g $HOST_GID $HOST_USERNAME || true
-RUN useradd -m -u $HOST_UID -g $HOST_GID $HOST_USERNAME || true
+RUN groupadd -g $HOST_GID $HOST_USERNAME
+RUN useradd -m -u $HOST_UID -g $HOST_GID $HOST_USERNAME

 # Install any dependencies we need in the container.
 ${ installDependencies( 'wordpress', env, config ) }`;
@@ -167,7 +167,7 @@ RUN adduser -h /home/$HOST_USERNAME -G $( getent group $HOST_GID | cut -d: -f1 )

 # Install any dependencies we need in the container.
 ${ installDependencies( 'cli', env, config ) }
-
+
 # Switch back to the original user now that we're done.
 USER www-data
```

When I run `npm run wp-env start`, I see that it fails to add groups and users as shown below. This problem does not occur on WSL2 (Linux).

```
#8 [ 5/17] RUN groupadd -g 0 tetsuaki.hamano
#8 0.286 groupadd: GID '0' already exists
#8 ERROR: process "/bin/sh -c groupadd -g $HOST_GID $HOST_USERNAME" did not complete successfully: exit code: 4
------
 > [ 5/17] RUN groupadd -g 0 tetsuaki.hamano:
#8 0.286 groupadd: GID '0' already exists
------
failed to solve: process "/bin/sh -c groupadd -g $HOST_GID $HOST_USERNAME" did not complete successfully: exit code: 4

```

Next, I thought the problem might be that `HOST_UID` and `HOST_GID` are zero and already exist. So I hard-coded each with a fixed value:

```diff
diff --git a/packages/env/lib/init-config.js b/packages/env/lib/init-config.js
index 4a573c97aa..efbed4486a 100644
--- a/packages/env/lib/init-config.js
+++ b/packages/env/lib/init-config.js
@@ -133,8 +133,8 @@ ARG HOST_USERNAME
 ARG HOST_UID
 ARG HOST_GID
 # When the IDs are already in use we can still safely move on.
-RUN groupadd -g $HOST_GID $HOST_USERNAME || true
-RUN useradd -m -u $HOST_UID -g $HOST_GID $HOST_USERNAME || true
+RUN groupadd -g 9999 $HOST_USERNAME || true
+RUN useradd -m -u 9999 -g 9999 $HOST_USERNAME || true

 # Install any dependencies we need in the container.
 ${ installDependencies( 'wordpress', env, config ) }`;
@@ -162,12 +162,12 @@ ARG HOST_USERNAME
 ARG HOST_UID
 ARG HOST_GID
 # When the IDs are already in use we can still safely move on.
-RUN addgroup -g $HOST_GID $HOST_USERNAME || true
-RUN adduser -h /home/$HOST_USERNAME -G $( getent group $HOST_GID | cut -d: -f1 ) -u $HOST_UID $HOST_USERNAME || true
+RUN addgroup -g 9999 $HOST_USERNAME || true
+RUN adduser -h /home/$HOST_USERNAME -G $( getent group 9999 | cut -d: -f1 ) -u 9999 $HOST_USERNAME || true

 # Install any dependencies we need in the container.
 ${ installDependencies( 'cli', env, config ) }
-
+
 # Switch back to the original user now that we're done.
 USER www-data
```

The error then changed to the following

```
npm run wp-env start

> gutenberg@15.8.1 wp-env D:\_local\WordPress\gutenberg\app\public\wp-content\plugins\gutenberg
> wp-env "start"

× Error while running docker-compose command.
 Container 9309c35b6b008e96c6c014f1a88dcbb4-mysql-1  Running
 Container 9309c35b6b008e96c6c014f1a88dcbb4-wordpress-1  Created
 Container 9309c35b6b008e96c6c014f1a88dcbb4-wordpress-1  Starting
 Container 9309c35b6b008e96c6c014f1a88dcbb4-wordpress-1  Started
Error: YIKES! It looks like you're running this as root. You probably meant to run this as the user that your WordPress installation exists under.

If you REALLY mean to run this as root, we won't stop you, but just bear in mind that any code on this site will then have full control of your server, making it quite DANGEROUS.

If you'd like to continue as root, please run this again, adding this flag:  --allow-root

If you'd like to run it as the user that this site is under, you can run the following to become the respective user:

    sudo -u USER -i -- wp <command>


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gutenberg@15.8.1 wp-env: `wp-env "start"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gutenberg@15.8.1 wp-env script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\tetsuaki.hamano\AppData\Roaming\npm-cache\_logs\2023-05-20T14_38_40_952Z-debug.log
```

--- a/packages/env/CHANGELOG.md
+++ b/packages/env/CHANGELOG.md
@@ -17,6 +17,10 @@
 -   Support using double dashes in `wp-env run ...` to pass arguments that would otherwise be consumed by `wp-env`. For example, while normally `--help` would provide the `wp-env` help text, if you use `npx wp-env run cli php -- --help` you will see the PHP help text.
 -   Validate whether or not config options exist to prevent accidentally including ones that don't.
 
+### Bug fix
+
+-   Support Windows without requiring the use of WSL.
+
 ## 7.0.0 (2023-05-10)
 
 ### Breaking Change

--- a/packages/env/lib/commands/run.js
+++ b/packages/env/lib/commands/run.js
@@ -67,8 +67,9 @@ function spawnCommandDirectly( config, container, command, envCwd, spinner ) {
 	// to interact with the files mounted from the host.
 	const hostUser = getHostUser();
 
-	// We need to pass absolute paths to the container.
-	envCwd = path.resolve(
+	// Since Docker requires absolute paths, we should resolve the input to a POSIX path.
+	// This is needed because Windows resolves relative paths from the C: drive.
+	envCwd = path.posix.resolve(
 		// Not all containers have the same starting working directory.
 		container === 'mysql' || container === 'tests-mysql'
 			? '/'

--- a/packages/env/lib/config/parse-config.js
+++ b/packages/env/lib/config/parse-config.js
@@ -87,6 +87,7 @@ const DEFAULT_ENVIRONMENT_CONFIG = {
 	testsPort: 8889,
 	mappings: {},
 	config: {
+		FS_METHOD: 'direct',
 		WP_DEBUG: true,
 		SCRIPT_DEBUG: true,
 		WP_ENVIRONMENT_TYPE: 'local',
@@ -233,7 +234,10 @@ async function getDefaultConfig(
 		env: {
 			development: {},
 			tests: {
-				config: { WP_DEBUG: false, SCRIPT_DEBUG: false },
+				config: {
+					WP_DEBUG: false,
+					SCRIPT_DEBUG: false,
+				},
 			},
 		},
 	};

--- a/packages/env/lib/config/test/parse-config.js
+++ b/packages/env/lib/config/test/parse-config.js
@@ -40,6 +40,7 @@ const DEFAULT_CONFIG = {
 	pluginSources: [],
 	themeSources: [],
 	config: {
+		FS_METHOD: 'direct',
 		WP_DEBUG: true,
 		SCRIPT_DEBUG: true,
 		WP_ENVIRONMENT_TYPE: 'local',

--- a/packages/env/lib/get-host-user.js
+++ b/packages/env/lib/get-host-user.js
@@ -13,14 +13,10 @@ module.exports = function getHostUser() {
 	const hostUser = os.userInfo();
 
 	// On Windows the uid and gid will be -1. Since there isn't a great way to handle this,
-	// we're just going to say that the host user is root. On Windows you'll likely be
-	// using WSL to run commands inside the container, which has a uid and gid. If
-	// you aren't, you'll be mounting directories from Windows, to a Linux
-	// VM (Docker Desktop uses one), to the guest OS. I assume that
-	// when dealing with this configuration that file ownership
-	// has the same kind of magic handling that macOS uses.
-	const uid = ( hostUser.uid === -1 ? 0 : hostUser.uid ).toString();
-	const gid = ( hostUser.gid === -1 ? 0 : hostUser.gid ).toString();
+	// we're just going to assign them to 1000. Docker Desktop already takes care of
+	// permission-related issues using magic, so this should be fine.
+	const uid = ( hostUser.uid === -1 ? 1000 : hostUser.uid ).toString();
+	const gid = ( hostUser.gid === -1 ? 1000 : hostUser.gid ).toString();
 
 	return {
 		name: hostUser.username,

--- a/packages/env/lib/init-config.js
+++ b/packages/env/lib/init-config.js
@@ -207,8 +207,13 @@ RUN echo "$HOST_USERNAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
 		}
 		case 'cli': {
 			dockerFileContent += `
+# Make sure we're working with the latest packages.
 RUN apk update
+
+# Install some basic PHP dependencies.
 RUN apk --no-cache add $PHPIZE_DEPS && touch /usr/local/etc/php/php.ini
+
+# Set up sudo so they can have root access.
 RUN apk --no-cache add sudo linux-headers
 RUN echo "$HOST_USERNAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
 			break;
@@ -238,7 +243,7 @@ RUN rm /tmp/composer-setup.php`;
 	// Install any Composer packages we might need globally.
 	// Make sure to do this as the user and ensure the binaries are available in the $PATH.
 	dockerFileContent += `
-USER $HOST_USERNAME
+USER $HOST_UID:$HOST_GID
 ENV PATH="\${PATH}:/home/$HOST_USERNAME/.composer/vendor/bin"
 RUN composer global require --dev yoast/phpunit-polyfills:"^1.0"
 USER root`;

--- a/packages/env/lib/init-config.js
+++ b/packages/env/lib/init-config.js
@@ -244,7 +244,7 @@ RUN rm /tmp/composer-setup.php`;
 	// Make sure to do this as the user and ensure the binaries are available in the $PATH.
 	dockerFileContent += `
 USER $HOST_UID:$HOST_GID
-ENV PATH="\${PATH}:/home/$HOST_USERNAME/.composer/vendor/bin"
+ENV PATH="\${PATH}:~/.composer/vendor/bin"
 RUN composer global require --dev yoast/phpunit-polyfills:"^1.0"
 USER root`;
 

--- a/packages/env/lib/init-config.js
+++ b/packages/env/lib/init-config.js
@@ -202,7 +202,7 @@ RUN apt-get -qy install $PHPIZE_DEPS && touch /usr/local/etc/php/php.ini
 
 # Set up sudo so they can have root access.
 RUN apt-get -qy install sudo
-RUN echo "$HOST_USERNAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
+RUN echo "$( whoami ) ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
 			break;
 		}
 		case 'cli': {
@@ -215,7 +215,7 @@ RUN apk --no-cache add $PHPIZE_DEPS && touch /usr/local/etc/php/php.ini
 
 # Set up sudo so they can have root access.
 RUN apk --no-cache add sudo linux-headers
-RUN echo "$HOST_USERNAME ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
+RUN echo "$( whoami ) ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
 			break;
 		}
 		default: {

--- a/packages/env/lib/init-config.js
+++ b/packages/env/lib/init-config.js
@@ -202,7 +202,7 @@ RUN apt-get -qy install $PHPIZE_DEPS && touch /usr/local/etc/php/php.ini
 
 # Set up sudo so they can have root access.
 RUN apt-get -qy install sudo
-RUN echo "$( whoami ) ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
+RUN echo "#$HOST_UID ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
 			break;
 		}
 		case 'cli': {
@@ -215,7 +215,7 @@ RUN apk --no-cache add $PHPIZE_DEPS && touch /usr/local/etc/php/php.ini
 
 # Set up sudo so they can have root access.
 RUN apk --no-cache add sudo linux-headers
-RUN echo "$( whoami ) ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
+RUN echo "#$HOST_UID ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers`;
 			break;
 		}
 		default: {

